source/common/upstream/health_checker_impl.cc

```diff
@@ -153,14 +155,25 @@ bool HttpHealthCheckerImpl::HttpStatusChecker::inRange(uint64_t http_status) con
   return false;
 }
 
+Http::Protocol codecClientTypeToProtocol(Http::CodecClient::Type codec_client_type) {
+  switch (codec_client_type) {
+  case Http::CodecClient::Type::HTTP1:
+    return Http::Protocol::Http11;
+  case Http::CodecClient::Type::HTTP2:
+    return Http::Protocol::Http2;
+  case Http::CodecClient::Type::HTTP3:
+    return Http::Protocol::Http3;
```

Comment:
Can we NOT_IMPLEMENTED this? Or will this fail somewhere else?

Reply:
If we config health check to use HTTP3, it will fail to create client codec as NOT_IMPLEMENTED.